paper_framework_sources/p1898_forward_progress_delegation.md

```diff
@@ -274,103 +285,87 @@ vector<int> doWork(Sender<vector<int>> vec) {
 # Impact on the Standard
  * Add `scheduler_provider` concept.
  * Add `get_scheduler` CPO.
- * Additional wording to the `sync_wait` CPO specifying that it will pass an `executor_provider` to the `submit` method of the passed `sender`.
- * Add `on` CPO to apply the `scheduler` both upstream and downstream.
- * Ensure that all asynchronous algorithms propagate the scheduler from the output to the input when this makes sense.
+ * Additional wording to the `sync_wait` CPO specifying that it will pass an `scheduler_provider` to the `connect` operation on the passed `sender`.
+ * Additional wording to the `on` CPO specifying that it will pass an `scheduler_provider` to the `connect` operation on the passed `sender`.
+ * Additional wording for other algorithms specifying that they pass through the `get_scheduler` CPO, making their own `receiver`s also `scheduler_provider`s.
 
 ## Concept scheduler_provider
 ### Summary
 A concept for receivers that may provide a scheduler upstream.
-May be used to overload submit methods to allow delegation in the presence of a downstream scheduler.
-May be used to restrict the submit method on a `sender` to require an `executor_provider`.
-This restriction would be important to implement folly's `SemiFuture` or `SemiAwaitable` concepts where all work is delegated.
+May be used to overload `connect` algorithm, to allow delegation in the presence of a downstream scheduler.
+May be used to restrict the `connect` operation on a `sender` to require an `scheduler_provider`.
 
 ### Wording
 ```
 template<class R>
 concept scheduler_provider =
   receiver<R> &&
```

Comment:
Were you going to relax the restriction for the argument to be a receiver?

Reply:
Oh, I thought I did. I obviously only half-completed that change.